llvm/include/llvm/Object/SYCLBIN.h

```diff
@@ -0,0 +1,126 @@
+//===- SYCLBIN.h - SYCLBIN binary format support ----------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_OBJECT_SYCLBIN_H
+#define LLVM_OBJECT_SYCLBIN_H
+
+#include "llvm/ADT/SmallString.h"
+#include "llvm/SYCLPostLink/ModuleSplitter.h"
+#include "llvm/Support/MemoryBuffer.h"
+#include <string>
+
+namespace llvm {
+
+namespace object {
+
+// Representation of a SYCLBIN binary object. This is intended for use as an
+// image inside a OffloadBinary.
+class SYCLBIN {
+public:
+  SYCLBIN(MemoryBufferRef Source) : Data{Source} {}
+
+  SYCLBIN(const SYCLBIN &Other) = delete;
```

Comment:
Would we like to have move constructors and move operators?

Reply:
Yes! I've applied rule-of-5 to all relevant cases I could find.